include/beast/websocket/stream.hpp

```diff
@@ -627,6 +627,14 @@ class stream
         return wr_opcode_ == detail::opcode::text;
     }
 
+
+    /// Returns `true` if either the server or the client sent a close frame.
+    bool 
+    closed() const
+    {
+        return wr_close_ || rd_close_;
```

Comment:
This will be unnecessary soon. If `rd_close_` is ever true, it means that `websocket::error::closed` has already been delivered to the caller. If `wr_close_` is true, it means the caller invoked `close` or `async_close`. And soon, those functions will also drain the connection and deliver `websocket::error::closed` to the caller to indicate a proper closure. So this function won't make much sense.